parsers/CA_AB.py

```diff
@@ -1,161 +1,153 @@
-            'solar': float(total_net_generation['SOLAR']),
-            'wind': float(total_net_generation['WIND']),
-            'biomass': float(total_net_generation['OTHER']),
-            'unknown': float(total_net_generation['DUAL FUEL']),
-            'coal': float(total_net_generation['COAL'])
-        },
-        'storage': {
-            'battery': float(total_net_generation['ENERGY STORAGE'])
-        },
-        'capacity': {
-            'gas': float(maximum_capability['GAS']),
-            'hydro': float(maximum_capability['HYDRO']),
-            'battery storage': float(maximum_capability['ENERGY STORAGE']),
-            'solar': float(maximum_capability['SOLAR']),
-            'wind': float(maximum_capability['WIND']),
-            'biomass': float(maximum_capability['OTHER']),
-            'unknown': float(maximum_capability['DUAL FUEL']),
-            'coal': float(maximum_capability['COAL'])
-        },
-        'source': 'ets.aeso.ca',
+        'datetime': get_csd_report_timestamp(response.text),
+        'sortedZoneKeys': sorted_zone_keys,
+        'netFlow': float(flows[sorted_zone_keys]),
+        'source': URL.netloc,
     }
 
 
-def fetch_price(zone_key='CA-AB', session=None, target_datetime=None, logger=None) -> dict:
-    """Requests the last known power price of a given country."""
+def fetch_price(zone_key=DEFAULT_ZONE_KEY,
+                session=None,
+                target_datetime=None,
+                logger=None) -> dict:
+    """Request the last known power price of a given country."""
     if target_datetime:
-        raise NotImplementedError('This parser is not yet able to parse past dates')
-
-    r = session or requests.session()
-    url = 'http://ets.aeso.ca/ets_web/ip/Market/Reports/SMPriceReportServlet?contentType=html/'
-    response = r.get(url)
-
-    df_prices = pd.read_html(response.text, match='Price', index_col=0, header=0)
-    prices = df_prices[1]
-
-    data = {}
-
-    for rowIndex, row in prices.iterrows():
-        price = row['Price ($)']
-        if (isfloat(price)):
-            hour = int(rowIndex.split(' ')[1]) - 1
-            data[rowIndex] = {
-                'datetime': arrow.get(rowIndex, 'MM/DD/YYYY').replace(hour=hour, tzinfo=ab_timezone).datetime,
-                'zoneKey': zone_key,
-                'currency': 'CAD',
-                'source': 'ets.aeso.ca',
-                'price': float(price),
-            }
-
-    return [data[k] for k in sorted(data.keys())]
-
-
-def fetch_exchange(zone_key1='CA-AB', zone_key2='CA-BC', session=None, target_datetime=None, logger=None) -> dict:
-    """Requests the last known power exchange (in MW) between two countries."""
+        raise NotImplementedError('Currently unable to scrape historical data')
+    session = session or requests.Session()
+    response = session.get(f'{URL_STRING}/SMPriceReportServlet',
+                           params={'contentType': 'csv'})
+    return [
```

Comment:
The API serves the hourly prices in reverse-chronological order; I'm not sure if they should be reversed, or whether the order matters at all.

Reply:
I couldn't tell from looking at the backend either, did you check how other parsers does it? I guess the ENTSO-E or FR should be trustworthy 🤔

Reply:
ENTSO-E and FR appear to return values in chronological (i.e., most recent last) order; however, neither parser explicitly sorts the results; the order is just a consequence of how the API presents the data. I tried shuffling the list with `random.shuffle`, and `poetry run test_parser CA-AB price` was successful no matter what order was presented to it. This leads me to believe that the order doesn't matter.

Reply:
It probably doesn't, but thanks for looking into it :)